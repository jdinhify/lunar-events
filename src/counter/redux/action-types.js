export const COUNTER_DECREMENT = 'COUNTER_DECREMENT'
export const COUNTER_INCREMENT = 'COUNTER_INCREMENT'
