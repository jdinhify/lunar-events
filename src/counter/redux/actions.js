import { createAction } from 'redux-actions'

import {
    COUNTER_DECREMENT,
    COUNTER_INCREMENT
} from './action-types'


export const counterIncrement = createAction(COUNTER_INCREMENT)
export const counterDecrement = createAction(COUNTER_DECREMENT)
