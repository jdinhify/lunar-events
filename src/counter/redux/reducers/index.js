import { handleActions } from 'redux-actions'

import {
    COUNTER_DECREMENT,
    COUNTER_INCREMENT
} from '../action-types'
import counterDecrementReducer from './counter-decrement'
import counterIncrementReducer from './counter-increment'

export const initialState = {
    value: 0
}

export const counterReducer =  handleActions({
    [COUNTER_DECREMENT]: counterDecrementReducer,
    [COUNTER_INCREMENT]: counterIncrementReducer
}, initialState)

export default counterReducer
