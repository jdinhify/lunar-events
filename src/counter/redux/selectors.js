export const counterValueSelector = state => state.counter.value
