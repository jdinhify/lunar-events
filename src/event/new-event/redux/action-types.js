export const EVENT_DETAIL_UPDATE = 'EVENT_DETAIL_UPDATE'
export const EVENT_DETAIL_CLEAR = 'EVENT_DETAIL_CLEAR'
