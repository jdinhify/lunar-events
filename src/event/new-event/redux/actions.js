import { createAction } from 'redux-actions'
import {
    EVENT_DETAIL_UPDATE,
    EVENT_DETAIL_CLEAR
} from './action-types'

export const updateEventDetail = ({ name, value }) => ({
    type: EVENT_DETAIL_UPDATE,
    name,
    value
})

export const clearEventDetail = createAction(EVENT_DETAIL_CLEAR)
