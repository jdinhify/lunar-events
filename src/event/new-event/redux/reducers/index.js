import { handleActions } from 'redux-actions'

import {
    EVENT_DETAIL_UPDATE,
    EVENT_DETAIL_CLEAR
} from '..'
import eventDetailUpdateReducer from './event-detail-update'
import eventDetailClearReducer from './event-detail-clear'

export const initialState = {
    description: '',
    lunarDay: '',
    lunarMonth: ''
}

export const newEventReducer = handleActions({
    [EVENT_DETAIL_UPDATE]: eventDetailUpdateReducer,
    [EVENT_DETAIL_CLEAR]: eventDetailClearReducer(initialState)
}, initialState)

export default newEventReducer
