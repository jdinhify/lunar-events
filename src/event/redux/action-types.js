export const EVENT_YEAR_UPDATE = 'EVENT_YEAR_UPDATE'
