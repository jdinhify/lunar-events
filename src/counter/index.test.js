import React from 'react'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'

import Counter from './'
import { initialState } from './redux'

const middleware = [thunk]
const mockStore = configureMockStore(middleware)

describe('Counter component', () => {
    it('renders correctly', () => {
        const store = mockStore({ counter: initialState })
        const component = renderer.create(
            <Counter store={store}/>
        )

        expect(component).toMatchSnapshot()
    })
})
