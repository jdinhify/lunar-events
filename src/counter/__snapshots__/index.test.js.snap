exports[`Counter component renders correctly 1`] = `
<div>
  <h1
    className="counter-heading">
    Counter
  </h1>
  <span
    className="counter-span">
    0
  </span>
  <button
    onClick={[Function]}>
    -
  </button>
  <button
    onClick={[Function]}>
    +
  </button>
</div>
`;
